slinger/src/main/java/pl/allegro/android/slinger/SlingerActivity.java

```diff
@@ -4,46 +4,50 @@
 import android.content.Intent;
 import android.net.Uri;
 import android.os.Bundle;
+
+import java.util.Collections;
+
 import pl.allegro.android.slinger.enricher.DefaultIntentEnricher;
 import pl.allegro.android.slinger.enricher.IntentEnricher;
 import pl.allegro.android.slinger.resolver.IntentResolver;
 import pl.allegro.android.slinger.resolver.RedirectRule;
-import java.util.Collections;
 
-public abstract class SlingerActivity extends Activity {
+public class SlingerActivity extends Activity {
 
-  private IntentEnricher intentEnricher = new DefaultIntentEnricher();
+  private static IntentEnricher intentEnricher = new DefaultIntentEnricher();
 
   @Override protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
 
-    Uri uri = getOriginatingUriFromIntent(getIntent());
+    startActivity(this, getIntent());
+    finish();
+  }
+
+  public static void startActivity(Activity parentActivity, Intent intent) {
```

Comment:
static startActivity is a bit confusing.
It's nice to have possibiliy of starting local activities using Slinger's deeplinking resolver, but maybe move it to some utility class (`Slinger`?).